tensorflow_io/core/kernels/azfs_kernels.cc

```diff
@@ -608,109 +692,189 @@ Status AzBlobFileSystem::DeleteDir(const std::string &dirname) {
-        blob_client.get_blob_property(target_container, target_object);
-    copy_status = target_blob_property.copy_status;
+    const auto dst_blob_property =
+        blob_client.get_blob_property(dst_container, dst_object);
+    copy_status = dst_blob_property.copy_status;
   } while (copy_status.find("pending") == 0 && !copy_status.empty());
 
   if (copy_status.find("success") == std::string::npos) {
-    return errors::Internal("Process of renaming resulted in status of ",
-                            copy_status, " when renaming ", src, " to ",
-                            target);
+    std::string error_message =
+        absl::StrCat("Process of renaming resulted in status of ", copy_status,
+                     " when renaming ", src, " to ", dst);
+    TF_SetStatus(status, TF_INTERNAL, error_message.c_str());
+    return;
   }
 
   blob_client.delete_blob(src_container, src_object);
   if (errno != 0) {
-    return errors::Internal("Failed to get delete after copy of ", src, " (",
-                            errno_to_string(), ")");
+    std::string error_message =
+        absl::StrCat("Failed to get delete after copy of ", src, " (",
+                     errno_to_string(), ")");
+    TF_SetStatus(status, TF_INTERNAL, error_message.c_str());
+    return;
   }
 
-  return Status::OK();
+  TF_SetStatus(status, TF_OK, "");
+}
+
+static void CopyFile(const TF_Filesystem* filesystem, const char* src,
+                     const char* dst, TF_Status* status) {
+  // 128KB copy buffer
+  constexpr size_t kCopyFileBufferSize = 128 * 1024;
+
+  std::string src_account, src_container, src_object;
+  ParseAzBlobPath(src, false, &src_account, &src_container, &src_object,
+                  status);
+  if (TF_GetCode(status) != TF_OK) {
+    return;
+  }
+  std::unique_ptr<AzBlobRandomAccessFile> src_file(
+      new AzBlobRandomAccessFile(src_account, src_container, src_object));
+
+  std::string dst_account, dst_container, dst_object;
+  ParseAzBlobPath(dst, false, &dst_account, &dst_container, &dst_object,
+                  status);
+  if (TF_GetCode(status) != TF_OK) {
+    return;
+  }
+  std::unique_ptr<AzBlobWritableFile> dst_file(
+      new AzBlobWritableFile(dst_account, dst_container, dst_object));
+
+  uint64_t offset = 0;
+  std::unique_ptr<char[]> buffer(new char[kCopyFileBufferSize]);
+  while (TF_GetCode(status) == TF_OK) {
+    int64_t bytes_to_read =
+        src_file->Read(offset, kCopyFileBufferSize, buffer.get(), status);
+    if (!(TF_GetCode(status) == TF_OK ||
+          TF_GetCode(status) == TF_OUT_OF_RANGE)) {
+      return;
+    }
+    dst_file->Append(buffer.get(), bytes_to_read, status);
+    if (TF_GetCode(status) != TF_OK) {
+      return;
+    }
+    offset += bytes_to_read;
+  }
+  dst_file->Close(status);
 }
 
-Status AzBlobFileSystem::RecursivelyCreateDir(const string &dirname) {
-  return CreateDir(dirname);
+static void PathExists(const TF_Filesystem* filesystem, const char* path,
```

Comment:
Shouldn't this return a `bool`?

Reply:
similar to `static bool IsDirectory()`

Reply:
@kvignesh1420 The function signature is defined by C modular system API so we will  not be able to change it (return bool).